src/finalisers/system.ts

```diff
@@ -109,8 +111,9 @@ export default function system (
 			`\n${t}${varOrConst} _starExcludes = { ${Array.from(starExcludes).join(': 1, ')}${starExcludes.size ? ': 1' : ''} };`;
 
 	const importBindingsSection = importBindings.length ? `\n${t}var ${importBindings.join(', ')};` : '';
+	const registeredName = outputOptions.name ? `'${outputOptions.name}', ` : '';
```

Comment:
Looking through rollup's [documentation](https://rollupjs.org/guide/en#big-list-of-options), I think using `name` here is an abuse of this option which is meant to define global variable names for IIFE/umd bundles (maybe we want to add SystemJS to our umd output at some point?). As @guybedford pointed out, the `amd.id` option would be more equivalent except for the `amd` part of the name. So I would prefer deprecating `amd.id` in favour of a new output option. Not sure what the best name could be;  we could call it something like `moduleId` as it is the id of the generated module as used by the loader. Maybe @guybedford has a more elaborate opinion on this.